coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -876,6 +890,113 @@ protected SqlNode visitCTE(ASTNode node, ParseContext ctx) {
     return result;
   }
 
+  @Override
+  protected SqlNode visitWindowSpec(ASTNode node, ParseContext ctx) {
+    // See Apache Hive source code: https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java#L4339
+    // "private RelNode genSelectForWindowing(QB qb, RelNode srcRel, HashSet<ColumnInfo> newColumns)"
+
+    // Hive Antlr ASTNode Tree:
+    //TOK_FUNCTION
+    //   MIN
+    //   TOK_TABLE_OR_COL
+    //   TOK_WINDOWSPEC  <-- processed by this node
+    //      TOK_PARTITIONINGSPEC
+    //         TOK_DISTRIBUTEBY
+    //         TOK_ORDERBY
+    //      TOK_WINDOWRANGE
+    //         CURRENT
+    //         FOLLOWING
+    //            1
+
+    // Calcite SqlNode Tree:
+    // SqlBasicCall(Over):
+    // - SqlBasicCall(ROW_NUMBER)
+    // - SqlWindow  <-- returned by this node
+    // -- partitionList
+    // -- orderList
+    // -- isRows
+    // -- lowerBound
+    // -- upperBound
+    // -- allowPartial
+
+    // Use a separate context to avoid inadvertently having the "ORDER BY" clause from the window added to the SELECT query.
+    ctx = new ParseContext(ctx.hiveTable.orElse(null));
+
+    SqlWindow partitionSpec = (SqlWindow) visitOptionalChildByType(node, ctx, HiveParser.TOK_PARTITIONINGSPEC);
+    SqlWindow windowRange = (SqlWindow) visitOptionalChildByType(node, ctx, HiveParser.TOK_WINDOWRANGE);
+    SqlWindow windowValues = (SqlWindow) visitOptionalChildByType(node, ctx, HiveParser.TOK_WINDOWVALUES);
+    SqlWindow window = windowRange != null ? windowRange : windowValues;
+
+    return new SqlWindow(ZERO, null, null, partitionSpec == null ? null : partitionSpec.getPartitionList(),
+        partitionSpec == null ? null : partitionSpec.getOrderList(),
+        SqlLiteral.createBoolean(windowRange != null, ZERO), window == null ? null : window.getLowerBound(),
+        window == null ? null : window.getUpperBound(), null);
+  }
+
+  @Override
+  protected SqlNode visitPartitioningSpec(ASTNode node, ParseContext ctx) {
+    SqlNode partitionList = visitOptionalChildByType(node, ctx, HiveParser.TOK_DISTRIBUTEBY);
+    SqlNode orderList = visitOptionalChildByType(node, ctx, HiveParser.TOK_ORDERBY);
+    return new SqlWindow(ZERO, null, null, (SqlNodeList) partitionList, (SqlNodeList) orderList, null, null, null,
+        null);
+  }
+
+  @Override
+  protected SqlNode visitDistributeBy(ASTNode node, ParseContext ctx) {
+    return new SqlNodeList(visitChildren(node, ctx), ZERO);
+  }
+
+  @Override
+  protected SqlNode visitWindowRange(ASTNode node, ParseContext ctx) {
+    // Hive AST:
+    //      TOK_WINDOWRANGE  (ROWS ...)
+    //         CURRENT
+    //         FOLLOWING
+    //            1
```

Comment:
Is this an example? Better to write the example in SQL and follow the same example in all functions.

Reply:
Added SQL example at the top of the comment.